constraints/match.js

```diff
@@ -111,5 +111,93 @@ function hasMatch(sNode, pTree)
 		}
 		return false;
 	}
-	
+
+}
+
+// Match Max constraints:
+
+/* Same as hasMatch function above, except this only returns true if the
+* matching prosodic node is maximal:
+*/
+function hasMaxMatch(sNode, pTree){
+	 var sLeaves = getLeaves(sNode);
+	 markMinMax(pTree); //mark min and max on prosodic tree
+	 if(catsMatch(sNode.cat, pTree.cat) && sameIds(getLeaves(pTree), sLeaves) && pTree.isMax)
+	 // the current prosodic node is the match, both for category and for terminals, and is maximal
+	 {
+ 		return true;
+ 		}
+
+ 		// If the current prosodic node is NOT the match:
+
+ 		else if(!pTree.children || pTree.children.length===0){
+ 		// current node is terminal
+ 			return false;
+		}
+
+ 		else
+ 		//the current prosodic node is non-terminal (has children)
+ 		{
+ 			for(var i = 0; i < pTree.children.length; i++)
+ 			//check each child to see if the match exists in the subtree rooted in that child
+ 			{
+ 				var child = pTree.children[i];
+ 				if(hasMatch(sNode, child)){
```

Comment:
What ensures that the match is maximal in this case?

Reply:
copy and past error. called hasMaxMatch in latest commit